azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -731,6 +739,15 @@ func resourceArmLinuxVirtualMachineScaleSetUpdate(d *schema.ResourceData, meta i
 		update.Sku = sku
 	}
 
+	if features.VMSSExtensionsBeta() {
+		if d.HasChange("extension") {
+			updateInstances = true
+
+			extensionProfile := expandVirtualMachineScaleSetExtensions(d.Get("extension").([]interface{}))
+			updateProps.VirtualMachineProfile.ExtensionProfile = extensionProfile
+		}
+	}
```

Comment:
(not an issue, for info) since this is using `d.HasChange` this should support mixing both the inline and external resource here as required